toxcore/tox.c

```diff
@@ -4580,3 +4573,31 @@ bool tox_group_mod_kick_peer(const Tox *tox, uint32_t group_number, uint32_t pee
 
 #endif /* VANILLA_NACL */
 
+Tox_System *tox_system_new(void)
+{
+    const Tox_System default_system = tox_default_system();
+
+    Tox_System *sys = (Tox_System *)mem_alloc(default_system.mem, sizeof(Tox_System));
+
+    if (sys == nullptr) {
+        return nullptr;
+    }
+
+    *sys = default_system;
+
+    return sys;
+}
+
+void tox_system_kill(Tox_System *sys)
+{
+    // Use default system, because we used that to allocate the object, and mem
+    // may have been overridden in the meantime.
+    const Tox_System default_system = tox_default_system();
+
+    mem_delete(default_system.mem, sys);
+}
+
+const Tox_System *tox_get_system(Tox *tox)
+{
+    return &tox->sys;
```

Comment:
Should this assert(tox != nullptr) like we do in the other functions taking a `Tox *`?

Reply:
Done.